src/main/java/graphics/Window.java

```diff
@@ -55,182 +50,150 @@ public Window(int pwidth, int pheight, String ptitle, boolean fullscreen, float
-        glfwWindowHint(GLFW_RESIZABLE, GLFW_TRUE);
-
-        initWindow(width, height, title, glfwGetPrimaryMonitor());
+        this(ptitle, false);
     }
 
-    private void initWindow (int width, int height, String title, long monitor) {
+    private void initWindow(int width, int height, String title, long monitor) {
         // Create window
-        window = glfwCreateWindow(width, height, title, monitor, 0);
+        glfwWindow = glfwCreateWindow(width, height, title, monitor, 0);
 
-        if (window == 0)
+        if (glfwWindow == 0)
             throw new IllegalStateException("[FATAL] Failed to create window.");
 
         // Set up callback
-        glfwSetWindowSizeCallback(window, (w, newWidth, newHeight) -> {
+        glfwSetWindowSizeCallback(glfwWindow, (w, newWidth, newHeight) -> {
             Window.setWidth(newWidth);
             Window.setHeight(newHeight);
 
-            if (recalculateProjectionOnResize && currentScene.camera != null) currentScene.camera.adjustProjection();
+            if (recalculateProjectionOnResize && currentScene().camera != null)
+                currentScene().camera.adjustProjection();
             Events.windowResizeEvent.onEvent(new EventData.WindowResizeEventData(newWidth, newHeight));
         });
 
         Mouse.setupCallbacks();
         Keyboard.setupCallbacks();
 
         // Make the OpenGL context current
-        glfwMakeContextCurrent(window);
+        glfwMakeContextCurrent(glfwWindow);
 
         // Enable V-Sync
 //        glfwSwapInterval(1);
 
         // Center the window
-        glfwSetWindowPos(window, (videoMode.width() - width) / 2, (videoMode.height() - height) / 2);
+        glfwSetWindowPos(glfwWindow, (videoMode.width() - width) / 2, (videoMode.height() - height) / 2);
         GL.createCapabilities();
+
     }
 
     void getFPS() {
+        //TODO this wont properly display the FPS it will just count up the frames, there is no reset after a second yet
         frameCount++;
-        glfwSetWindowTitle(window, title + " @ " + Math.round((frameCount / (Engine.millis() / 1000))) + " FPS");
+        glfwSetWindowTitle(glfwWindow, title + " @ " + Math.round((frameCount / (Engine.millisRunning() / 1000))) + " FPS");
+    }
+
+    public static long glfwWindow() {
+        return glfwWindow;
+    }
+
+    public String getTitle() {
+        return title;
     }
 
     public void setTitle(String title) {
-        glfwSetWindowTitle(window, title);
+        this.title = title;
+        glfwSetWindowTitle(glfwWindow, title);
     }
 
     public void showWindow() {
         /*
          * scenes.Main game loop
          */
-        glfwShowWindow(window);
+        glfwShowWindow(glfwWindow);
 
         glEnable(GL_BLEND);
         glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
 
-        float frameBeginTime = (float)glfwGetTime();
-        float frameEndTime = (float)glfwGetTime();
+        double frameBeginTime = glfwGetTime();
+        double frameEndTime;
```

Comment:
This change to double is fine for now, I do plan on redoing the delta time calculation later as the framerate calculation takes too long to accurately average out.